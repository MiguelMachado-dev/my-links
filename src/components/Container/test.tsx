import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Container } from '.'

describe('<Container />', () => {
  it('should render <Container /> correctly', () => {
    const { container } = render(
      <Container>
        <h2>Test</h2>
      </Container>
    )

    screen.logTestingPlaygroundURL()

    const heading = screen.getByRole('heading', {
      name: /test/i,
    })

    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toBeInTheDocument()

    expect(heading).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle({
      width: '100%',
      maxWidth: '60rem',
      marginLeft: 'auto',
      marginRight: 'auto',
      paddingLeft: '1.6rem',
      paddingRight: '1.6rem',
      paddingTop: '7.2rem',
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
