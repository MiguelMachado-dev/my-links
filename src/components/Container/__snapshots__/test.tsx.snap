// Vitest Snapshot v1

exports[`<Container /> > should render <Container /> correctly 1`] = `
<h2>
  Test
</h2>
`;

exports[`<Container /> > should render <Container /> correctly 2`] = `
.c0 {
  width: 100%;
  max-width: 60rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: 1.6rem;
  padding-right: 1.6rem;
  padding-top: 7.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <h2>
    Test
  </h2>
</div>
`;
